backend/algorithms/objects/conditions.py

```diff
@@ -342,7 +342,7 @@ def __init__(self, exclusion: str):
         self.exclusion = exclusion
 
     def validate(self, user: User) -> tuple[bool, list[str]]:
-        is_valid = not user.has_taken_course(self.exclusion)
+        is_valid = not user.has_taken_specific_course(self.exclusion)
```

Comment:
🤨

Reply:
yea, i know, wierd. Exclusions probably should be exhaustive, so i split off the two types.